specs/Standards/Storage.md

```diff
@@ -0,0 +1,203 @@
+# Storage Standard ([NEP-145](https://github.com/near/NEPs/discussions/145))
+
+Version `1.0.0`
+
+## Summary
+
+This standard handles state storage when accounts add and remove data stored in the state of a contract. It allows accounts to:
+
+1. Check an account's storage "balance."
+2. Determine the minimum storage needed to add account information such that the account can interact as expected in a contract.
+3. Add storage for an account; either their own or another account.
+4. Withdraw some or all of a storage deposit by removing associated account data from the contract and then making a call to remove unused deposit.
+
+## Motivation
+
+NEAR uses [storage staking](https://docs.near.org/docs/concepts/storage-staking) as the mechanism to pay for storage on the blockchain. When storage is removed, the staked Ⓝ is released. Developers must take this into consideration when crafting smart contracts that store data on-chain. If no restrictions are enforced in a permissionless system, any account may add key/value pairs that bloat another account's storage. Since an account containing a smart contract must keep enough Ⓝ to pay for the storage, developers must ensure the state storage is "paid for" as the storage increases, and can refund such storage deposits later if data is removed. This standard provides a generic approach to handle these concerns.
+
+Prior art:
+
+- A previous fungible token standard ([NEP-21](https://github.com/near/NEPs/pull/21)) highlighting how [storage was paid](https://github.com/near/near-sdk-rs/blob/1d3535bd131b68f97a216e643ad1cba19e16dddf/examples/fungible-token/src/lib.rs#L92-L113) for when increasing the allowance of an escrow system.
+
+## Guide-level explanation
+
+We should be able to do the following:
+
+- Have an account sign a transaction, attaching a deposit of Ⓝ that will cover *their own* storage for a smart contract.
+- Have an account sign a transaction, attaching a deposit of Ⓝ that will cover the storage for another account to use a smart contract.
+- Allow an account to remove all their associated data stored on a contract and receive a refund for that storage in the form of Ⓝ.
```

Comment:
Updating this to:

> Allow an account to request refunds in Ⓝ for excess storage deposits on a given contract.

Does that seem right to you, @oysterpack?

Reply:
The idea of withdrawing is removing all their data from the contract so they can wash their hands of any involvement in the contract.

Reply:
Seems there is disagreement here, with @oysterpack and @evgenykuzyakov both stating that this standard is not meant to cover account closure/removal. Is there a recorded call where this was discussed and an agreement was reached for what is in and out of scope for the Account Storage standard?

Reply:
`storage_withdraw` is only allowed to withdraw from the account's **available** storage balance. Thus, the total account storage balance cannot be withdrawn via `storage_withdraw` because the required minimum storage balance must be maintained.

To be clear,  this standard is not meant to cover account closure/removal - see [discussion][1] which documents the decision.

I would phrase as:
> Allow an account to withdraw Ⓝ from its storage available balance. The account storage balance is defined as the Ⓝ  amount in excess to the minimum account storage balance that is required by the contract.

and add an **Out of Scope** section for:
> - How to close the account and be able to withdraw all funds.
> - How the contract should account for changes in price for storage on the NEAR blockchain over time.

My [tutorial][2] also serves as a good reference.

[1]: https://github.com/near/NEPs/discussions/145#discussioncomment-361351
[2]: https://learn.figment.io/network-documentation/near/tutorials/1-project_overview/5-account-storage#quick-api-overview